core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -497,10 +502,17 @@ public boolean isMailerPluginPresent() {
 
     /**
      * Creates a new user account by registering a password to the user.
+     *
+     * Supports bcrypt hashed passwords if the password begins with the characters
+     * <code>#jbcrypt:</code>
      */
     public User createAccount(String userName, String password) throws IOException {
         User user = User.getById(userName, true);
-        user.addProperty(Details.fromPlainPassword(password));
+        if (password.startsWith(JBCRYPT_HEADER)) {
```

Comment:
By doing so, you are changing the expected behavior for some weird cases. I would prefer to have a separate method like `createAccountWithHashedPassword` if the goal is only for CasC.

In the current approach you are impacting users that desires to have a real password starting `#jbcrypt:`. I understand that the impact is quite small but such behavior could be potentially used by malicious users to achieve things.

Reply:
So I had thought of that approach as well. However, then the onus is on the caller to know if the password is hashed or not before calling the appropriate method (it also ensures that CASC now takes a hard dependency on a certain future version of Jenkins, which I'm unsure if that's desirable or not).

See the current implementation here: https://github.com/jenkinsci/configuration-as-code-plugin/pull/683, which is kind of messy, although can be done by making a couple of things public. I'm fine with whatever approach you think is better here though.

Reply:
I'm not very fond of encoding switching logic inside string messages unless there is a strong reason to do so. It ends up hiding important information and making it more difficult to figure out what is going on. With things like this I find it is generally preferable to make the client be explicit about which type of data it is passing rather than expecting the service to guess. Far too often there end up being cases in which the service guesses incorrectly.

Doesn't CASC have a strong dependency either way?

Reply:
I'm cool with that. How does the update look to you?